promql/engine.go

```diff
@@ -519,6 +521,18 @@ func (ng *Engine) populateSeries(ctx context.Context, q storage.Queryable, s *Ev
 	return querier, err
 }
 
+func extractGroupsFromPath(p []Node) []string {
```

Comment:
I don't think this is safe as-is, for example:

```
max by  (foo)(a - b)
```

Would fail to do proper label matching between a and b if a remote storage was using this. I think we need to limit this to aggregators either directly around the selector, or around a function which is around the selector.